awx/main/analytics/core.py

```diff
@@ -82,145 +61,152 @@ def decorate(f):
     return decorate
 
 
-def gather(dest=None, module=None, subset = None, since = None, until = now(), collection_type='scheduled'):
+def package(target, data, timestamp):
+    try:
+        tarname_base = f'{settings.SYSTEM_UUID}-{timestamp.strftime("%Y-%m-%d-%H%M%S%z")}'
+        path = pathlib.Path(target)
+        index = len(list(path.glob(f'{tarname_base}-*.*')))
+        tarname = f'{tarname_base}-{index}.tar.gz'
+
+        manifest = {}
+        with tarfile.open(target.joinpath(tarname), 'w:gz') as f:
+            for name, (item, version) in data.items():
+                manifest[name] = version
+                if isinstance(item, str):
+                    info = f.gettarinfo(item, arcname=name)
+                    f.addfile(info)
+                else:
+                    info = tarfile.TarInfo(name)
+                    fileobj = io.BytesIO(json.dumps(item).encode('utf-8'))
+                    fileobj.size = len(fileobj.getvalue())
+                    f.addfile(info, fileobj=fileobj)
+
+            info = tarfile.TarInfo('manifest.json')
```

Comment:
@jbradberry this might be a good place to add the config json? So it's ensured it's part of every payload?